kola/tests/misc/files.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         SUIDFiles,
+		ClusterSize: 1,
+		Name:        "coreos.filesystem.suid",
+		UserData:    `#cloud-config`,
+	})
+	register.Register(&register.Test{
+		Run:         SGIDFiles,
+		ClusterSize: 1,
+		Name:        "coreos.filesystem.sgid",
+		UserData:    `#cloud-config`,
+	})
+	register.Register(&register.Test{
+		Run:         WritableFiles,
+		ClusterSize: 1,
+		Name:        "coreos.filesystem.writablefiles",
+		UserData:    `#cloud-config`,
+	})
+	register.Register(&register.Test{
+		Run:         WritableDirs,
+		ClusterSize: 1,
+		Name:        "coreos.filesystem.writabledirs",
+		UserData:    `#cloud-config`,
+	})
+}
+
+func sugidFiles(m platform.Machine, validfiles []string, mode string) error {
+	badfiles := make([]string, 0, 0)
+
+	command := fmt.Sprintf("sudo find / -path /sys -prune -o -path /proc -prune -o -type f -perm -%v -print", mode)
+
+	output, err := m.SSH(command)
+	if err != nil {
+		return fmt.Errorf("Failed to run find: output %v, status: %v", output, err)
+	}
+
+	if string(output) == "" {
+		return nil
+	}
+
+	files := strings.Split(string(output), "\n")
+	for _, file := range files {
+		var valid bool
+
+		for _, validfile := range validfiles {
+			if file == validfile {
+				valid = true
+			}
+		}
+		if valid != true {
+			badfiles = append(badfiles, file)
+		}
+	}
+
+	if len(badfiles) != 0 {
+		return fmt.Errorf("Unknown SUID or SGID files found: %v", badfiles)
+	}
+
+	return nil
+}
+
+func SUIDFiles(c platform.TestCluster) error {
+	m := c.Machines()[0]
+
+	validfiles := []string{
```

Comment:
nti: For the sake of keeping future diffs clean lets start off with this list being sorted.